windows-gaming-driver/src/config.rs

```diff
@@ -11,10 +11,16 @@ pub struct DeviceId {
     pub product: u16,
 }
 
-#[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq, Eq, Copy)]
+pub struct DevicePort {
```

Comment:
Perhaps `UsbPort` is a better name. `DeviceId` is already kind of an ugly hack since it conflates USB's product+vendor with PCI's product+device IDs.